#26383 Refactor content type fetching and processing

jgambarios opened the pull request:
Refactored the `ContentTypeFetcher` to use iterator and recursive tasks for more efficient content type fetching and processing. Introduced `ContentTypeIterator` for paginated content type fetch, and `HttpRequestTask` for concurrent HTTP requests using Fork/Join framework. Additionally, minor changes made to `Relationship` and `Workflow` models.

jgambarios committed:
#26383 Refactor content type fetching and processing

Refactored the ContentTypeFetcher to use iterator and recursive tasks for more efficient content type fetching and processing. Introduced ContentTypeIterator for paginated content type fetch, and HttpRequestTask for concurrent HTTP requests using Fork/Join framework. Additionally, minor changes made to Relationship and Workflow models.

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-26383-CLI-Implement-a-global-pull-for-content-types-fixes

jgambarios committed:
#26383 Update relative path resolution in ContentTypePush and SitePush

Instead of resolving against the 'languages' path, they now resolve properly against the 'contentTypes' and 'sites' path respectively.

jgambarios committed:
Merge remote-tracking branch 'origin/master' into issue-26383-CLI-Implement-a-global-pull-for-content-types-fixes

fabrizzio-dotCMS commented on tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/pull/contenttype/HttpRequestTask.java:
```diff
@@ -0,0 +1,68 @@
+    @Override
+    protected List<ContentType> compute() {
+
+        if (contentTypes.size() <= THRESHOLD) {
```
I think we could have a more effective "divide and conquer" type of strategy here.
As it is now if we have only 10 then we do only one request that's totally fine... 
But I'm thinking in larger numbers let's say 250 CT  we could divide the list in bunches of 4 
or do something like 

```
final int partitionSize = Math
				.max((contentTypes.size() / maxThreads), 10);
```
then partition the list of ct 
and then divide the partition between a fixed number of tasks

fabrizzio-dotCMS replied:
only suggestion. its up to you

jgambarios committed:
#26383 Small refactor in the LanguagePush command

The most significant change is the removal of conditional checks and restructuring of the workspace verification process. Now, the command ensures the path is within a workspace and the file can be read before processing further, thus simplifying and streamlining the command execution.

jgambarios committed:
#26383 Update comments in ContentTypeFetcher.java

jgambarios committed:
#26383 Fixing language tests to properly use workspaces
